internal/pkg/admission/policy-server-deployment.go

```diff
@@ -125,6 +125,7 @@ func shouldUpdatePolicyServerDeployment(policyServer *policiesv1.PolicyServer, o
 		originalDeployment.Spec.Template.Spec.SecurityContext != newDeployment.Spec.Template.Spec.SecurityContext ||
 		originalDeployment.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation] != newDeployment.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation] ||
 		!reflect.DeepEqual(originalDeployment.Spec.Template.Spec.Containers[0].Env, newDeployment.Spec.Template.Spec.Containers[0].Env) ||
+		!reflect.DeepEqual(originalDeployment.Spec.Template.Spec.Containers[0].Resources, newDeployment.Spec.Template.Spec.Containers[0].Resources) ||
```

Comment:
This made me think... can we have 100% sure that the first container will be the policy server one? Cannot be the OPTEL collector sometimes?

If we are not sure about that, I think we should open an issue to investigate or maybe mitigate this.

Reply:
Good call, I would defer the changes here as we are probably going to refactor this by using controller-runtime CreateOrUpdate utilities in 1.13